src/main/java/com/lovetropics/minigames/common/core/game/behavior/instances/action/CountdownAction.java

```diff
@@ -58,7 +61,11 @@ public void register(IGamePhase game, EventRegistrar events) {
 	private boolean tickQueuedAction(IGamePhase game, QueueEntry entry) {
 		long remainingTicks = entry.time() - game.ticks();
 		if (remainingTicks <= 0) {
-			return actions.apply(game, entry.context(), entry.sources());
+			if (actions.target.type() == target.type()) {
+				return actions.applyIf((java.util.function.Supplier) target::type, game, entry.context, entry.sources);
```

Comment:
If `QueueEntry` captures `T` from `ActionTarget<?>` (as well as the target itself to bind `T`), the unchecked casts here can be avoided 🙂